config_option.go

```diff
@@ -0,0 +1,147 @@
+package gorm
+
+import (
+	"sync"
+	"time"
+
+	"gorm.io/gorm/clause"
+	"gorm.io/gorm/logger"
+	"gorm.io/gorm/schema"
+)
+
+// ConfigOption use functional option for gorm Config.
+type ConfigOption func(c *Config)
+
+// WithSkipDefaultTransaction enable SkipDefaultTransaction.
+func WithSkipDefaultTransaction() ConfigOption {
+	return func(c *Config) {
+		c.SkipDefaultTransaction = true
+	}
+}
+
+// WithNameStrategy set shema namer.
+func WithNameStrategy(namer schema.Namer) ConfigOption {
+	return func(c *Config) {
+		c.NamingStrategy = namer
+	}
+}
+
+// WithFullSaveAssociations set FullSaveAssociations = true.
+func WithFullSaveAssociations() ConfigOption {
+	return func(c *Config) {
+		c.FullSaveAssociations = true
+	}
+}
+
+// WithLogger set logger.
+func WithLogger(logger logger.Interface) ConfigOption {
+	return func(c *Config) {
+		c.Logger = logger
+	}
+}
+
+// WithNowFunc set now func.
+func WithNowFunc(fn func() time.Time) ConfigOption {
+	return func(c *Config) {
+		c.NowFunc = fn
+	}
+}
+
+// WithEnableDryRun enable dry run.
+func WithEnableDryRun() ConfigOption {
+	return func(c *Config) {
+		c.DryRun = true
+	}
+}
+
+// WithPrepareStmt enable PrepareStmt.
+func WithPrepareStmt() ConfigOption {
+	return func(c *Config) {
+		c.PrepareStmt = true
+	}
+}
+
+// WithEnableAutomaticPing enable ping.
+func WithEnableAutomaticPing() ConfigOption {
+	return func(c *Config) {
+		c.DisableAutomaticPing = true
+	}
+}
+
+// WithEnableForeignKeyConstraintWhenMigrating
+// ForeignKeyConstraintWhenMigrating config.
+func WithEnableForeignKeyConstraintWhenMigrating() ConfigOption {
+	return func(c *Config) {
+		c.DisableForeignKeyConstraintWhenMigrating = true
+	}
+}
+
+// WithEnableNestedTransaction enable NestedTransaction.
+func WithEnableNestedTransaction() ConfigOption {
+	return func(c *Config) {
+		c.DisableNestedTransaction = true
+	}
+}
+
+// WithAllowGlobalUpdate allow global update.
+func WithAllowGlobalUpdate() ConfigOption {
+	return func(c *Config) {
+		c.AllowGlobalUpdate = true
+	}
+}
+
+// WithEnableQueryFields open QueryFields.
+func WithEnableQueryFields() ConfigOption {
+	return func(c *Config) {
+		c.QueryFields = true
+	}
+}
+
+// WithCreateBatchSize set batch size.
+func WithCreateBatchSize(size int) ConfigOption {
+	return func(c *Config) {
+		c.CreateBatchSize = size
+	}
+}
+
+// WithClauseBuilders set clause builder.
+func WithClauseBuilders(m map[string]clause.ClauseBuilder) ConfigOption {
+	return func(c *Config) {
+		c.ClauseBuilders = m
+	}
+}
+
+// WithConnPool set conn pool.
```

Comment:
The following configs maybe not necessary?

Reply:
Here I want to use the option method to set the parameters by default when opening gorm DB, and then call the Apply method.Maybe I think it’s different from you, because the usage scenarios are different.